pyro/infer/mcmc/nuts.py

```diff
@@ -0,0 +1,135 @@
+from __future__ import absolute_import, division, print_function
+
+from collections import namedtuple
+
+import torch
+
+import pyro
+import pyro.distributions as dist
+from pyro.infer.util import torch_data_sum
+from pyro.ops.integrator import velocity_verlet
+from pyro.util import ng_ones, ng_zeros
+
+from .hmc import HMC
+
+
+_TreeInfo = namedtuple("TreeInfo", ["z_left", "r_left", "z_right", "r_right",
+                                    "z_proposal", "size", "turning", "diverging"])
+
+
+class NUTS(HMC):
+    """
+    No-U-Turn Sampler kernel, where ``step_size`` need to be explicitly specified by the user.
+
+    References
+
+    [1] `The No-U-turn sampler: adaptively setting path lengths in Hamiltonian Monte Carlo`,
+    Matthew D. Hoffman, and Andrew Gelman
+
+    :param model: python callable containing pyro primitives.
+    :param float step_size: determines the size of a single step taken by the
+        verlet integrator while computing the trajectory using Hamiltonian
+        dynamics.
+    """
+
+    def __init__(self, model, step_size=0.5):
+        super(NUTS, self).__init__(model, step_size, num_steps=None)
+        self._reset()
+        self._dE_max = 1000
```

Comment:
Add a comment explaining this attribute

Reply:
Yes, it is important so I will explain it in the implementation.